prompting/validators/reward/blacklist.py

```diff
@@ -140,11 +195,19 @@ def calculate_significance(self) -> dict:
 
         significance_scores = {}
         for ngram, count in self.counter.items():
-
-            # calculate significance score for ngram
-            significance_scores[ngram] = self.A ** (len(ngram) - 1) * (count - 1)
-
-        self._last_update = self._running_size
+            if count[0] + count[1] > max(
+                self.support * self.num_completion, self.w_current + 1
+            ):
+                decoded_ngram = self.tokenizer.decode(ngram)
+                if len(decoded_ngram.split()) >= self.n_min:
+                    # calculate significance score for ngram
+                    significance_scores[decoded_ngram] = (
+                        self.A ** (len(decoded_ngram) - 1)
+                        * ((count[0] + count[1]) / self.num_completion)
+                        * 100
```

Comment:
why the factor of 100?

Reply:
Just added a param 'frequency_multiplier' to store the param 100. 

It was set at 100 cause most of the frequencies that we are seeing range from 0.01 - 0.1, so it kind of just make sense to scale it 100 times to 1 - 10 for the score.